flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/TimeFunctionsITCase.java

```diff
@@ -580,7 +583,36 @@ private Stream<TestSetSpec> ceilTestCases() {
                                 $("f2").ceil(TimeIntervalUnit.MILLENNIUM),
                                 "CEIL(f2 TO MILLENNIUM)",
                                 LocalDateTime.of(3001, 1, 1, 0, 0),
-                                TIMESTAMP().nullable()));
+                                TIMESTAMP().nullable())
+                        .testResult(
+                                $("f2").cast(TIMESTAMP_LTZ(3))
+                                        .ceil(TimeIntervalUnit.HOUR)
+                                        .cast(STRING()),
+                                "CAST(CEIL(CAST(f3 AS TIMESTAMP_LTZ(3)) TO HOUR) AS STRING)",
+                                formatAsTimestamp(LocalDateTime.of(2020, 2, 29, 2, 0, 0, 0)),
+                                STRING().nullable())
+                        .testResult(
+                                $("f2").cast(TIMESTAMP_LTZ(3))
+                                        .ceil(TimeIntervalUnit.MINUTE)
+                                        .cast(STRING()),
+                                "CAST(CEIL(CAST(f2 AS TIMESTAMP_LTZ(3)) TO MINUTE) AS STRING)",
+                                formatAsTimestamp(LocalDateTime.of(2020, 2, 29, 1, 57, 0, 0)),
+                                STRING().nullable())
+                        .testResult(
+                                $("f3").cast(TIMESTAMP_LTZ(3))
+                                        .ceil(TimeIntervalUnit.SECOND)
+                                        .cast(STRING()),
+                                "CAST(CEIL(CAST(f3 AS TIMESTAMP_LTZ(3)) TO SECOND) AS STRING)",
+                                formatAsTimestamp(LocalDateTime.of(2020, 2, 29, 1, 56, 58, 0)),
```

Comment:
why here 58 seconds?
we are making ceil, it should go up
or did I miss anything?

Reply:
Because the LocalDataTime is `LocalDateTime.of(2020, 2, 29, 1, 56, 57, 924325471)`, so ceil second will become to 58.